src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseInsecureDeserializerBinaryFormatterBannedMethods.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.NetCore.Analyzers.Security.Helpers;
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    /// <summary>
+    /// For detecting deserialization with <see cref="System.Runtime.Serialization.Formatters.Binary.BinaryFormatter"/>.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    class DoNotUseInsecureDeserializerBinaryFormatterBannedMethods : DoNotUseInsecureDeserializerBannedMethodsBase
+    {
+        // TODO paulming: Help links URLs.
+        internal static readonly DiagnosticDescriptor RealBannedMethodDescriptor =
+            new DiagnosticDescriptor(
+                "CA2300",
+                GetResourceString(nameof(MicrosoftNetCoreSecurityResources.BinaryFormatterBannedMethodTitle)),
```

Comment:
>GetResourceString [](start = 16, length = 17)

Probably better to instead have a `CreateDescriptor(string id, string titleResourceName, ...)` static method on the base type?